pym/bob/input.py

```diff
@@ -1574,7 +1585,12 @@ def __init__(self, recipeSet, recipe, sourceFile, baseDir, packageName, baseName
             scm["__source"] = sourceName
             scm["recipe"] = "{}#{}".format(sourceFile, i)
             i += 1
-        self.__checkout = (checkoutScript, checkoutDigestScript, checkoutSCMs)
+        checkoutAsserts = recipe.get("checkoutAssert", [])
+        if isinstance(checkoutAsserts, dict):
+            checkoutAsserts = [checkoutAsserts]
+        elif not isinstance(checkoutAsserts, list):
+            raise ParseError("checkoutAssert must be a dict or a list")
```

Comment:
Shouldn't that be handled already by the schema check? BTW, the schema allows only a list anyway...